### PR TITLE
fix: Parsing of negative numbers in machine expressions

### PR DESCRIPTION
**Pull request by slavek-kucera**

### CODE REVIEW

File: parser_library/src/parsing/grammar/ca_expr_rules.g4 [hunk 1]
```diff
@@ -186,48 +186,59 @@ expr_comma_c returns [std::vector<ca_expr_ptr> ca_exprs]
...
-	{
-		collector.add_hl_symbol(token_info(provider.get_range( $NUM),hl_scopes::var_symbol));
-		$point = std::make_unique<char_str_conc>($NUM->getText(), provider.get_range($NUM));
-	}
-	| var_symbol											{$point = std::make_unique<var_sym_conc>(std::move($var_symbol.vs));}
-	| dot													{$point = std::make_unique<dot_conc>();};
-
-created_set_body_c returns [concat_chain concat_list]
-	: (cl=created_set_body {$concat_list.push_back(std::move($cl.point));})+;
+created_set_body returns [concat_chain concat_list]
```
Comment (jirimosinger) [inline parser_library/src/parsing/grammar/ca_expr_rules.g4:189]: Consider appending `_c` to maintain consistency with other methods returning `chain`s

File: parser_library/src/parsing/grammar/ca_expr_rules.g4 [hunk 2]
```diff
@@ -303,19 +328,16 @@ ca_dupl_factor returns [ca_expr_ptr value]
...
-		$value.substring_range = provider.get_range($lpar.ctx->getStart(), $rpar.ctx->getStop());
-		resolve_expression($value.start, context::SET_t_enum::A_TYPE);
-		resolve_expression($value.count, context::SET_t_enum::A_TYPE);
-	}
-	| lpar expr comma ASTERISK rpar
-	{
-		$value.start = std::move($expr.ca_expr);
 		$value.substring_range = provider.get_range($lpar.ctx->getStart(), $rpar.ctx->getStop());
 		resolve_expression($value.start, context::SET_t_enum::A_TYPE);
+		if ($e2.ctx)
```
Comment (jirimosinger) [inline parser_library/src/parsing/grammar/ca_expr_rules.g4:336]: Did you mean `$e2.ca_expr`?

Comment (slavek-kucera) [reply parser_library/src/parsing/grammar/ca_expr_rules.g4:336]: The `$e2.ctx` tests the presence of the `expr` parsing context (aliased by `e2` label). This tests which path (expression or asterisk) the parsed had chosen. The `$e2.ca_expr` does not exist unless `$e2.ctx != nullptr`.

Comment (jirimosinger) [reply parser_library/src/parsing/grammar/ca_expr_rules.g4:336]: 👍

File: parser_library/src/parsing/grammar/ca_expr_rules.g4 (hunk 1 above)
Comment (jirimosinger) [reply parser_library/src/parsing/grammar/ca_expr_rules.g4:189]: Alright

**Review by jirimosinger — APPROVED** [review]

**Pull request merged**